content/blog/eng/from-zero-to-infinite-the-final-keyword.md

```diff
@@ -16,45 +16,47 @@ toc: true
+-- [PHP: Final Keyword - Manual](https://www.php.net/manual/en/language.oop5.final.php)
 
-First of all we will try to understand what it is for, then we will try to do some additional reasoning, and finally we will talk about of comments on the web.
+Java and C# use the same keyword but it is called **sealed** instead of `final`. These keywords have more or less the same behavior: they prevent a class from being inherited.  In this article, we will discuss this bahavior only, even if some languages like Java support further behaviors in different contexts.
 
-The concept that expresses the keyword final is that a class can not be extended, the decision must be taken by the programmer in the design stage utilizing just the **final**. 
+In this article we will start by explaining the purpose of the final keyword. Then, we will try to do some additional reasoning. Finally, we will talk about comments found on the net.
 
-There are situations in which extension by inheritance is a good solution, others where it is not possible and others where it is wrong. The keyword **final** is a "tool" given to the programmer to express this constraint.
+The concept expressed by the final keyword is that a class cannot be extended. The programmer must take this decision at the design stage by using this keyword.
 
-Why someone should set such a strong constraint?. The answers are many, all aim of having a robust, and more manageable code-base. For example to avoid the proliferation of infinite hierarchies of classes such as VCL (https://en.wikipedia.org/wiki/Visual_Component_Library), which are difficult to extend and maintain. In this case, even being a well-made library, the use of inheritance has been used a lot. 
+In some cases, extension by inheritance is a good solution while in some others it is either not possible or wrong. The keyword `final` is a "tool" given to the programmer to express this constraint.
 
-Also I use final is to reduce the API surface that needs to be covered by BC breaks avoidance in a distributed library.
+Why should someone set such a strong constraint?. The answers are many, all aim of having a robust, and more manageable code-base. For example to avoid the proliferation of infinite hierarchies of classes such as VCL (https://en.wikipedia.org/wiki/Visual_Component_Library), which are difficult to extend and maintain. In this case, even being a well-made library, the use of inheritance has been used a lot. 
 
-Further to use **final** in some cases forces adopt the interfaces, by testing for example, this is a positive side effect.
+Furthermore, I use the `final` keyword to reduce the API surface API surface that needs to be covered by BC breaks avoidance in a distributed library.
```

Comment:
Furthermore, I use the `final` keyword to reduce the API surface that needs to be covered by BC breaks avoidance in a distributed library.

Nota: c'era  API surface ripetuto due volte